src/goto-programs/goto_trace.cpp

```diff
@@ -420,7 +420,14 @@ void show_goto_trace(
       break;
 
     case goto_trace_stept::typet::FUNCTION_CALL:
+      if(options.show_function_calls)
+        out << "\n####  Function call: " << as_string(ns, *step.pc)
```

Comment:
Cheap and easy improvement: track the stack so you can say what function is returned from, and mention the stack depth at each call/return (e.g. `Function call: f (depth 3)`, and later `Function return: f (depth 3)`)

Reply:
Somewhat harder, don't expect to do in this PR but worth considering: option to indent the trace steps accordingly